docs/proposals/networking/multiclusterservice.md

```diff
@@ -0,0 +1,184 @@
+- [multicluster-cloud-provider](https://github.com/karmada-io/multicluster-cloud-provider) repo provides the shared interfaces for the MultiClusterService to connect to the cloud provider LoadBalancer service.
+
+### Non-Goals
+
+Implementation of the interconnection with the LoadBalancer service by a specific cloud provider.
+
+## Proposal
+
+### User Stories (Optional)
+
+#### Story 1
+
+As end users, to ensure HA and improve cluster fault tolerance, users need to deploy the same service in different clusters in multiple regions and expose the service to the LB service of the cloud provider. In this way, during access, requests can be evenly distributed to backend instances in different clusters.
+
+#### Story 2
+
+As end users, when using the MCS feature to discovery service between clusters, they need to create multiple resources: ServiceExport, ServiceImport, PropagationPolicy, which is complex. If unified API is provided, the usage threshold is lowered.
+
+### Notes/Constraints/Caveats (Optional)
+
+### Risks and Mitigations
+
+## Design Details
+
+### General Idea
+
+1. Provides MultiClusterService API, users can define MultiClusterService resource to expose service on the Karmada control-plane.
+2. The type field in the MultiClusterService API to provide the corresponding type value so that the current atomic capabilities can be integrated and an overall **inter-clusters service discovery** solution can be provided through the MultiClusterService API.
+
+### MultiClusterService API
+
+```go
+type MultiClusterService struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec is the desired state of the MultiClusterService.
+	// +optional
+	Spec ServiceSpec `json:"spec,omitempty"`
+
+	// Status is the current state of the MultiClusterService.
+	// +optional
+	Status corev1.ServiceStatus `json:"status,omitempty"`
+}
+
+type ServiceSpec struct {
+	// Backend specifies the exposed service backend. 
+	// Only valid for LoadBalance ServiceType.
+	// +optional
+	Backend *ServiceBackend `json:"backend"`
+
+	// Types specifies the including types.
+	//
+	// When includes LoadBalance ServiceType, the service is exposed externally.
+	// When includes ClusterSetIP ServiceType, provides inter-clusters service discovery.
+	// The source cluster set is the cluster set of the service,
+	// the destination cluster set is specified by ImportClusters.
+	// 
+	// When multiple types are specified, the capabilities corresponding to each type are provided.
+	// Can not be empty.
+	Types []ServiceType `json:"types"`
+
+	// ImportClusters specifies the destination cluster set to which services need to be imported.
+	// Only valid only for ClusterSetIP ServiceType.
+	// If not set, all clusters are selected.
+	// +optional
+	ImportClusters *ImportClusters `json:"importClusters"`
+}
+
+type ServiceBackend struct {
+	// Port specifies the exposed service port.
+	Port int32 `json:"port"`
```

Comment:
What is the difference between `Port` here and the `Ports` in `ServiceImport`?

https://github.com/kubernetes/enhancements/blob/master/keps/sig-multicluster/1645-multi-cluster-services-api/README.md#importing-services

Reply:
Port in ServiceBackend are used to expose services.
Ports in ServiceImport will be filled based on the ports info of the target Service when ServiceImport is automatically created.

Reply:
I update the port to ports.